r/tests/testthat/latin1.R

```diff
@@ -35,12 +35,12 @@ expect_identical(Encoding(levels(df[[2]])), "latin1")
 # Array
 expect_identical(as.vector(Array$create(x)), x)
 # struct
-expect_identical(as.vector(Array$create(df)), df)
+expect_identical(as.data.frame(Array$create(df)), df)
```

Comment:
This is a change in interface, so we might also want to also make the backwards compatibility option change this behavior too.  The new behavior seems more right here (especially in a world where not all structs are dataframes), but we might want to be more careful than this.